clang/lib/Frontend/FrontendAction.cpp

```diff
@@ -945,6 +945,7 @@ bool FrontendAction::BeginSourceFile(CompilerInstance &CI,
 
     // Set the shared objects, these are reset when we finish processing the
     // file, otherwise the CompilerInstance will happily destroy them.
+    CI.setVirtualFileSystem(AST->getVirtualFileSystemPtr());
```

Comment:
Although `CI.getVirtualFileSystemPtr()` is being passed into `ASTUnit::LoadFromASTFile()`, the `AST` may end up being configured with a different VFS compared to `CI` which trips the assertion in `CompilerInstance::setFileManager()`. This is a good opportunity to fix that.